### ISSUE TITLE
Ability to create custom networks with options such as --internal flag

### ISSUE THREAD

**Issue by jhovell** [fabric8io/docker-maven-plugin#867]
### Description
From the documentation it's possible to autoCreateCustomNetworks which is very helpful. Is it possible to specify any additional configuration when automatically creating (and deleting) these networks, such as specifying the `--internal` flag for restricting access to internet.

https://docs.docker.com/engine/reference/commandline/network_create/#options

### Proposed idea
Would be great to have a global block such as the following to specify default custom network options.

```xml
<autoCreateCustomNetworkOptions>
  <internal>true</internal>
  ...
</autoCreateCustomNetworkOptions>
```

### Use Case
Using DMP for testing we would like to guarantee that our application has properly mocked out 3rd party services (typically via additional Docker containers in the same custom network) which could be contacted over the custom network, but also check that the application is not making calls to 3rd party dependencies outside the test environment.

Having to abandon using autoCreateCustomNetworks and manage network creation/removal on our own just to set this one flag seems unfortunate + require a lot of extra management.

**Comment by rhuss** [comment]
I agree that this feature makes sense. For the documentation format suggest to introduce a new configuration structure (replacing the single `autoCreateCustomNetworks` property):

```xml
<configuration>
...
  <networks>
     <autoCreate>true</autoCreate>
     <!-- Options which are applied to every created network, including auto created networks -->
     <options>
         <internal>true</internal>
          ....
     </options>
     <!-- Eplicitely created network, with name. These are created in <start> and destroyed in <stop> (like for autoCreated networks) -->
     <network>
          <name>net1</name>
          <!-- Network specific options, overwriting global options -->
          <options>
           ....
          </options>
      </network>
      <network> ... </network>
      ....
  </networks>
</configuration>
```
(Remark: Such 'mixed' configs (without an intermedate list level) can be achieved in Maven like described in http://blog.sonatype.com/2011/03/configuring-plugin-goals-in-maven-3/) 

I know this is much more involved but would provide a simple upgrade path to a better network handling. Also, it doesn't add to the configuration space (which since the old config `autoCreateCustomNetworks` will eventually vanish after being depricated). We could even start with the global `<options>` sections, allowing specific custom networks to be configured in a later step.

Wdyt ?

Unfortunately I'm a bit busy these days (changes hopefully a bit towards the end of the year), but if you feel fancy and want to work on it, I happily would review and integrate a pull request for this.
